test/PushComm/unit_tests/SendingNotifications/SendNotifsViaSig.t.sol

```diff
@@ -0,0 +1,209 @@
+    }
+
+    function test_WhenInvokedByAMaliciousUserUsingSignatureReplay() external {
+        // it Returns false on signature replay
+        bytes32 DOMAIN_SEPARATOR = getDomainSeparator();
+        SendNotif memory _sendNotif = SendNotif(
+            address(verifierContract),
+            actor.alice_channel_owner,
+            _testChannelIdentity,
+            commProxy.nonces(address(verifierContract)),
+            block.timestamp + 1000
+        );
+        bytes32 structHash = getStructHash(_sendNotif);
+        bytes32 digest = keccak256(abi.encodePacked("\x19\x01", DOMAIN_SEPARATOR, structHash));
+        (uint8 v, bytes32 r, bytes32 s) = vm.sign(privateKeys[actor.tim_push_holder], digest);
+        //Notification already sent one time
+        commProxy.sendNotifBySig(
+            address(verifierContract),
+            actor.alice_channel_owner,
+            address(verifierContract),
+            _testChannelIdentity,
+            commProxy.nonces(address(verifierContract)),
+            block.timestamp + 1000,
+            v,
+            r,
+            s
+        );
+
+        //Using the same signature twice should fail
+        bool res = commProxy.sendNotifBySig(
+            address(verifierContract),
+            actor.alice_channel_owner,
+            address(verifierContract),
+            _testChannelIdentity,
+            commProxy.nonces(address(verifierContract)),
+            block.timestamp + 1000,
+            v,
+            r,
+            s
+        );
+        assertEq(false, res);
+    }
+
+    function test_WhenInvokedByAMaliciousUserUsingExpiredSignature() external {
+        // it Returns false on signature expire
+        bytes32 DOMAIN_SEPARATOR = getDomainSeparator();
+        SendNotif memory _sendNotif = SendNotif(
+            address(verifierContract),
+            actor.alice_channel_owner,
+            _testChannelIdentity,
+            commProxy.nonces(address(verifierContract)),
+            block.timestamp + 1000
+        );
+        bytes32 structHash = getStructHash(_sendNotif);
+        bytes32 digest = keccak256(abi.encodePacked("\x19\x01", DOMAIN_SEPARATOR, structHash));
+        (uint8 v, bytes32 r, bytes32 s) = vm.sign(privateKeys[actor.tim_push_holder], digest);
+
+        bool res = commProxy.sendNotifBySig(
+            address(verifierContract),
+            actor.alice_channel_owner,
+            address(verifierContract),
+            _testChannelIdentity,
+            commProxy.nonces(address(verifierContract)),
+            block.timestamp - 1,
+            v,
+            r,
+            s
+        );
+
+        assertEq(false, res);
+    }
+
+    //HELPER FUNCTIONS
+    struct SendNotif {
+        address _channel;
+        address _recipient;
+        bytes _identity;
+        uint256 nonce;
+        uint256 expiry;
+    }
+
+    // computes the hash of a sendNotif
+    function getStructHash(SendNotif memory _sendNotif) internal pure returns (bytes32) {
```

Comment:
There could be a BasePushCommTest file that could have these extra helper fn and variables